File: Exercism Codes/Python/rail_fence_cipher.py
def encode(message, rails):
    return ''.join(message[x] for x in create_fence(len(message), rails))


def decode(encoded_message, rails):
    fence = create_fence(len(encoded_message), rails)
    return ''.join(encoded_message[fence.index(x)] for x in range(len(encoded_message)))


def create_fence(msg_len, rails):
    cycle = 2 * rails - 2
    return flatten([[y for y in range(msg_len) if y % cycle == x or y % cycle == cycle - x] for x in range(rails)])


def flatten(lst):
    return [item for sublist in lst for item in sublist]

File: Exercism Codes/Java/RailFenceCipher.java
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class RailFenceCipher {
    private final int rails;

    public RailFenceCipher(int rails) {
        this.rails = rails;
    }

    public String getEncryptedData(String input) {
        return createFence(input.length())
                .stream()
                .map(x -> String.valueOf(input.charAt(x)))
                .collect(Collectors.joining());
    }

    public String getDecryptedData(String input) {
        List<Integer> fence = createFence(input.length());
        return IntStream.range(0, input.length())
                .mapToObj(x -> String.valueOf(input.charAt(fence.indexOf(x))))
                .collect(Collectors.joining());
    }

    private List<Integer> createFence(int inputLength) {
        final int cycle = 2 * rails - 2;
        return IntStream.range(0, rails)
                .flatMap(x -> IntStream.range(0, inputLength)
                        .filter(y -> y % cycle == x || y % cycle == cycle - x))
                .boxed()
                .toList();
    }
}